src/syntax/binding_to_syntax.ts

```diff
@@ -27,68 +34,59 @@ class BindingToSyntax<T> implements interfaces.BindingToSyntax<T> {
     }
 
     public toConstantValue(value: T): interfaces.BindingWhenOnSyntax<T> {
-        this._binding.type = BindingTypeEnum.ConstantValue;
-        this._binding.cache = value;
-        this._binding.dynamicValue = null;
-        this._binding.implementationType = null;
-        this._binding.scope = BindingScopeEnum.Singleton;
-        return new BindingWhenOnSyntax<T>(this._binding);
+        return this.initialize("toConstantValue", value,true);
     }
 
     public toDynamicValue(func: interfaces.DynamicValue<T>): interfaces.BindingInWhenOnSyntax<T> {
-        this._binding.type = BindingTypeEnum.DynamicValue;
-        this._binding.cache = null;
-        this._binding.dynamicValue = func;
-        this._binding.implementationType = null;
-        return new BindingInWhenOnSyntax<T>(this._binding);
+        return this.initialize("toDynamicValue",func, false);
     }
 
     public toConstructor<T2>(constructor: interfaces.Newable<T2>): interfaces.BindingWhenOnSyntax<T> {
-        this._binding.type = BindingTypeEnum.Constructor;
-        this._binding.implementationType = constructor as any;
-        this._binding.scope = BindingScopeEnum.Singleton;
-        return new BindingWhenOnSyntax<T>(this._binding);
+        //tbd change generics
```

Comment:
Mentioned in https://github.com/inversify/InversifyJS/issues/1319#issue-872365711